src/tables/math.rs

```diff
@@ -0,0 +1,1062 @@
+    /// For shapes growing in the horizontal direction.
+    pub horiz_glyph_construction: LazyOffsetArray16<'a, MathGlyphConstruction<'a>>,
+}
+
+impl<'a> FromSlice<'a> for MathVariants<'a> {
+    fn parse(data: &'a [u8]) -> Option<Self> {
+        let mut s = Stream::new(data);
+        let min_connector_overlap = s.read()?;
+        let vert_glyph_coverage = s.parse_at_offset16(data)?;
+        let horiz_glyph_coverage = s.parse_at_offset16(data)?;
+        let vert_glyph_count = s.read()?;
+        let horiz_glyph_count = s.read()?;
+        let vert_glyph_construction_offsets = s.read_array16(vert_glyph_count)?;
+        let horiz_glyph_construction_offsets = s.read_array16(horiz_glyph_count)?;
+        Some(MathVariants {
+            min_connector_overlap,
+            vert_glyph_coverage,
+            horiz_glyph_coverage,
+            vert_glyph_construction: LazyOffsetArray16::new(data, vert_glyph_construction_offsets),
+            horiz_glyph_construction: LazyOffsetArray16::new(data, horiz_glyph_construction_offsets),
+        })
+    }
+}
+
+/// The `MathGlyphConstruction` table provides information on finding or assembling extended
+/// variants for one particular glyph. It can be used for shapes that grow in either horizontal or
+/// vertical directions.
+///
+/// Note that it is quite possible that both the [`GlyphAssembly`] table and some variants are
+/// defined for a particular glyph. For example, the font may provide several variants of curly
+/// braces with different sizes, and also a general mechanism for constructing larger versions of
+/// curly braces by stacking parts found in the glyph set. First, an attempt is made to find glyph
+/// among provided variants. If the required size is larger than any of the glyph variants
+/// provided, however, then the general mechanism can be employed to typeset the curly braces as a
+/// glyph assembly.
+#[allow(missing_docs)]
+#[allow(missing_debug_implementations)]
+#[derive(Clone)]
+pub struct MathGlyphConstruction<'a> {
+    pub glyph_assembly: Option<GlyphAssembly<'a>>,
+    pub variants: LazyArray16<'a, MathGlyphVariantRecord>,
+}
+
+impl<'a> FromSlice<'a> for MathGlyphConstruction<'a> {
+    fn parse(data: &'a [u8]) -> Option<Self> {
+        let mut s = Stream::new(data);
+        let glyph_assembly = s.parse_at_offset16(data);
+        let variant_count = s.read()?;
+        let variants = s.read_array16(variant_count)?;
+        Some(MathGlyphConstruction { glyph_assembly, variants })
+    }
+}
+
+/// Description of math glyph variants.
+#[derive(Debug, Default, Copy, Clone)]
+pub struct MathGlyphVariantRecord {
+    /// Glyph ID for the variant.
+    pub variant_glyph: GlyphId,
+    /// Advance width/height, in design units, of the variant, in the direction of requested glyph
+    /// extension.
+    pub advance_measurement: u16,
+}
+
+impl FromData for MathGlyphVariantRecord {
+    const SIZE: usize = 4;
```

Comment:
Please add a new line after each `const` in each `FromData`. This is probably the only formatting change I would love to see.

Reply:
Edited all five occurrences.